src/Orchard.Web/Modules/Orchard.OutputCache/Filters/OutputCacheFilter.cs

```diff
@@ -117,53 +121,44 @@ public class OutputCacheFilter : FilterProvider, IActionFilter, IResultFilter, I
                 if (allowServeFromCache && cacheItem != null) {
 
                     Logger.Debug("Item '{0}' was found in cache.", _cacheKey);
+                    
+                    if(cacheItem.IsValid(_now)) {
+                        Logger.Debug("Cached item is valid: {0}", _cacheKey);
+                        ServeCachedItem(filterContext, cacheItem);
+                        return;
+                    }
+                    else {
+                        Logger.Debug("Cached item is invalid: {0}", _cacheKey);
 
-                    // Is the cached item in its grace period?
-                    if (cacheItem.IsInGracePeriod(_now)) {
-
-                        // Render the content unless another request is already doing so.
-                        if (Monitor.TryEnter(_cacheKey)) {
+                        // If the cached item is in its grace period 
+                        // and there are no current renderer, this request renders the new content
```

Comment:
Nitpicking again but the comment needs punctuation at the end and there is a grammatical error: either "are ... renderers" or "is ... renderer". Judging from the context the latter one.